pkg/channel/distributed/dispatcher/dispatcher/dispatcher.go

```diff
@@ -177,9 +178,15 @@ func (d *DispatcherImpl) UpdateSubscriptions(subscriberSpecs []eventingduck.Subs
 			}
 
 		} else {
-
 			// Otherwise Just Add To List Of Active Subscribers
 			activeSubscriptions[subscriberSpec.UID] = true
+
+			// If the group is stopped, it's still active but the reconciler needs to know about it in order
+			// to not treat it as a failure (which would re-create the group, effectively un-stopping it)
+			if d.consumerMgr.IsStopped(groupId) {
+				d.Logger.Debug("Adding Stopped ConsumerGroup To Stopped Map", zap.String("GroupId", groupId))
+				stoppedSubscriptions[subscriberSpec.UID] = struct{}{}
+			}
```

Comment:
Still reviewing but... I'm wondering why the `stoppedSubscriptions` has a type of struct and really whether the separate map is needed at all?  The only different treatment of the two maps (failed/stopped) is that we hard-code an error for the stopped ones.  Couldn't we put then all in the same map and just hardcode the stopped message here instead?

Reply:
I looked into combining the maps before, but not in that way (I was trying to combine all three maps).  I'll give it another try with only those two maps and see what happens (I can't "just have the error" unless I compare the error against the hard-coded value to see if it's stopped vs. failed after reconcile() calls UpdateSubscriptions -- right now if "length of failedSubscriptions > 0" then reconcile() returns an error, which we don't want if all of our "failures" are actually just stopped)

Reply:
ok, i missed that nuance - don't have to force it into one array - i thought the only use was to know to hardcode the error message.

Reply:
I refactored this using a new type of "SubscriberStatusMap" and I think I've gotten it a fair bit cleaner now, to the point at least where this confusion wouldn't have happened :)